internal/pkg/pluginmanager/manager.go

```diff
@@ -25,37 +27,49 @@ func DownloadPlugins(conf *configloader.Config) error {
 
 	for _, tool := range conf.Tools {
 		pluginFileName := configloader.GetPluginFileName(&tool)
+		pluginMD5FileName := configloader.GetPluginMD5FileName(&tool)
+		// plugin does not exist
 		if _, err := os.Stat(filepath.Join(pluginDir, pluginFileName)); errors.Is(err, os.ErrNotExist) {
-			// plugin does not exist
-			err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version)
-			if err != nil {
+			// download .so file
+			if err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version); err != nil {
+				return err
+			}
+			// download .md5 file
+			if err := dc.download(pluginDir, pluginMD5FileName, tool.Plugin.Version); err != nil {
+				return err
+			}
+			// check if the downloaded plugin md5 matches with .md5
+			if err := checkPluginMismatch(pluginDir, pluginFileName, pluginMD5FileName, tool.Name); err != nil {
 				return err
 			}
 			continue
 		}
-		// check md5
-		dup, err := checkFileMD5(filepath.Join(pluginDir, pluginFileName), dc, pluginFileName, tool.Plugin.Version)
+
+		// if .so exists
```

Comment:
What if *.so exists but *.md5 doesn't?

Reply:
updated,
if no *.so, download it;
if no *.md5, download it;
then compare them;
if match, contiue;
if not match, re-download(remove and download both .so and .md5)
comapre them
if not match, raise error